.github/workflows/ci-rerun-trigger.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  issue_comment:
+    types:
+      - created
+
+name: Continuous Integration Re-Trigger
+jobs:
+  trigger:
+    runs-on: ubuntu-latest
+    name: Trigger
+    steps:
+      - name: Trigger
+        uses: actions/github-script@v6
+        if: github.event.issue.pull_request && contains(github.event.comment.body, '@eth-bot rerun')
+        with:
+          script: |
+            let pr = await github.rest.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.payload.issue_comment.issue.number
+            });
+            await github.rest.pulls.update({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.payload.issue_comment.issue.number,
+              body: '[RETRIGGER]\n' + pr.data.body
+            });
+            await github.rest.pulls.update({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.payload.issue_comment.issue.number,
+              body: pr.data.body
+            });
```

Comment:
So this re-runs the workflow by editing the body of the PR?

Is there some way to use [`workflow_dispatch`](https://octokit.github.io/rest.js/v18#actions-create-workflow-dispatch) for this? Or perhaps [re-running a check](https://docs.github.com/en/rest/checks/runs#rerequest-a-check-run)?

If you've already investigated these options, sorry!

Reply:
Re-running a check and workflow_dispatch requires write permission to use, which you have stated has too many permissions.

Reply:
It looks like something like this might work, if you can find the right run id:

```javascript
await github.request('POST /repos/{owner}/{repo}/actions/runs/{run_id}/rerun', {
  owner: context.repo.owner,
  repo: context.repo.repo,
  run_id: '...'
});
```

I tried it with an account without write permissions on my repo, so maybe `issue_comment` gets elevated permissions?

Reply:
Sorry, I should have specified: I thought you were talking about the editors clicking the button.

I'll see what I can do there.

Reply:
So that would be ideal, but GitHub doesn't have an API for fetching check runs!

Reply:
That's ridiculous...

Can we open/close the PR instead of editing it?

Reply:
Yes... but that pollutes history.

The reason I suggest editing the body is that it's pretty much invisible.

Reply:
Makes sense!